### PR TITLE
Github action with multi-CPU architecture support and multi docker file support

### PR DESCRIPTION
**Pull request by skabashnyuk**
- Build and push to docker registry Dockerfile under linux/amd64,linux/s390x,linux/arm64,linux/ppc64le
 - Build and push to docker registry rhel.Dockerfile under linux/amd64
 - Allow to configure with secrets DOCKER_USERNAME, DOCKER_PASSWORD, DOCKER_IMAGE, DOCKER_REPO
 - Same workflow for master and prs
 - Is able to push tags

Results https://quay.io/repository/skabashn/che-jwtproxy?tab=tags
Synced code with https://github.com/che-incubator/configbump/pull/5

### CODE REVIEW

**Comment by skabashnyuk** [comment]
CC @nickboldt @mkuznyetsov @fbricon @kumarvikas2020

File: build/dockerfiles/Dockerfile [hunk 1]
```diff
@@ -10,19 +10,76 @@
...
+ARG BUILD_DATE
+ARG VCS_REF
+ARG VERSION
+
+ARG TARGETPLATFORM
+ARG BUILDPLATFORM
+RUN printf "I am running on ${BUILDPLATFORM:-linux/amd64}, building for ${TARGETPLATFORM:-linux/amd64}\n$(uname -a)\n" \
+  && $(case ${TARGETPLATFORM:-linux/amd64} in \
+      "linux/amd64")   echo "GOOS=linux GOARCH=amd64" > /tmp/.env                       ;; \
+      "linux/arm/v6")  echo "GOOS=linux GOARCH=arm GOARM=6" > /tmp/.env                 ;; \
```
Comment (benoitf) [inline build/dockerfiles/Dockerfile:24]: you may drop armv6 v7 and 386 I think ?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
+name: build
```
Comment (benoitf) [inline .github/workflows/build.yml:1]: you may add copyright headers

**Comment by benoitf** [comment]
It is leaving rhel.Dockerfile untouched is it still working ?

**Comment by benoitf** [comment]
should all cico scripts be disabled/removed as well ? it won't collide ?

**Comment by skabashnyuk** [comment]
@benoitf 
>It is leaving rhel.Dockerfile untouched is it still working ?

Yes, I've checked. It works.

>should all cico scripts be disabled/removed as well ? 

Don't want to hurry up. Let's live it for some time.

>it won't collide ?

Don't know. Need to talk with @mkuznyetsov @nickboldt

**Comment by nickboldt** [comment]
+0. I'm not going to comment on a PR that changes 417 files in a repo for which I'm neither an owner not an expert. If you want my input please make this into a set of several smaller, more manageable PRs.

**Comment by skabashnyuk** [comment]
@nickboldt it has 5 changed files and removed vendor folder are you sure you want me to split it?

**Comment by nickboldt** [comment]
No, I'm OK with the deleted files since you explained the rationale in Slack. Upgrading my vote to +0.5

But I'm still wondering if you've forgotten to do anything in the rhel.Dockerfile file to match your changes in Dockerfile.

https://github.com/eclipse/che-jwtproxy/blob/a2fcdaedd8dfa4d660e23c5e7b17c07749d22acb/build/dockerfiles/rhel.Dockerfile

If you can explain why these changes are not applicable to downstream (as I suspect is the case), or add changes to the rhel.Dockerfile too, then I'll be +1.

**Comment by skabashnyuk** [comment]
@nickboldt 

 
-  https://github.com/eclipse/che-jwtproxy/pull/30/files#diff-5d421a2eae745dc052d563fba3467293R21-R30 that is about multi-CPU architecture support which is irrelevant to rhel
-  Alpine update  -irrelevant to rhel
https://github.com/eclipse/che-jwtproxy/pull/30/files#diff-5d421a2eae745dc052d563fba3467293R57-R66
 https://github.com/eclipse/che-jwtproxy/pull/30/files#diff-5d421a2eae745dc052d563fba3467293R70-R74
- Image labels. Might be relevant. https://github.com/eclipse/che-jwtproxy/pull/30/files#diff-5d421a2eae745dc052d563fba3467293R57-R66

The core and the significant part I believe are identical.

**Pull request merged**